extension/js/common/core/attachment.ts

```diff
@@ -199,10 +199,10 @@ export class Attachment {
       // ends with one of .gpg, .pgp, .???.asc, .????.asc
       return 'encryptedFile';
       // todo: after #4906 is done we should "decrypt" the encryptedFile here to see if it's a binary 'publicKey' (as in message 1869220e0c8f16dd)
+    } else if (this.name.match(/flowcrypt-backup-[a-z0-9]+-[A-F0-9]{40}\.(key|asc)$/g)) {
+      return 'privateKey';
     } else if (this.isPublicKey()) {
       return 'publicKey';
-    } else if (this.name.match(/(cryptup|flowcrypt)-backup-[a-z0-9]+\.(key|asc)$/g)) {
```

Comment:
I updated the regex name as the FlowCrypt backup name is no longer having the prefix `cryptup`.

Reply:
But some old backups still has `cryptup` in their names, so correct parsing will stop working for them.

Also I have my key backup from 2021 with name `flowcrypt-backup-<MY_EMAIL>.key` and parsing stopped working for it, as new regex expects key fingerprint after `<MY_EMAIL>`. Regex should work for cases when fingerprint isn't included in backup name

Reply:
Ah, yes. I will update the regex so that it can cover both old and new regex. Thanks again for your help.